control-plane/gateways/deployment.go

```diff
@@ -36,22 +36,26 @@ func (b *meshGatewayBuilder) deploymentSpec() (*appsv1.DeploymentSpec, error) {
 		return nil, err
 	}
 
-	var containerConfig *meshv2beta1.GatewayClassContainerConfig
+	var (
+		containerConfig *meshv2beta1.GatewayClassContainerConfig
+		nodeSelector    map[string]string
```

Comment:
I wonder if this should be perhaps be changed to `nodeSelector = map[string]string{}`, because if b.gcc is empty, I believe this will panic as opposed to setting the value as nil.

Reply:
I _think_ it should work because you can assign a nil value to the field in the deployment and it should "just work :tm:" but I'm going to add the test just  to be sure

Reply:
so yeah, this would panic if we were doing anything more than assigning the variables to a field on the deployment struct, if we tried accessing anything in them is where we'd have that nil panic issue

Reply:
Nice! Good catch